Mackbellemore/soen-390-team07
# feat(server): sales endpoints

#

## Changes this PR introduces

- Create a sale, get all sales, update a sale's status endpoints



## How to test this PR

- Make a `POST` request at the `/sales` endpoint with the following body:
```json
{
	"bikeId": "603726757a92b8001347072b",
	"customerEmail": "any@email.com",
	"quantity": 1,
	"customer": "You",
	"status": "Placed"
}
```
Note that the `bikeId` needs to be a valid bike found in the database
- Make a `PATCH` request to the `/sales` endpoint with the previous sale document ID:
```json
  {
    "status": "Cancelled",
    "_id": "________",
  }
```
You should see the bike stock go back up and the new sale's status at the `GET` endpoint of `/sales`



## ClickUp reference

CU-n32713

## Notes for reviewers

- When a sale is placed, the bike stock is decremented
- When a sale is cancelled, the bike stock is incremented 



## Checklist
